### PR TITLE
test InsufficientFlowOutputs revert and boundary

### LINKED ISSUE #311: [A02-2] [MEDIUM] InsufficientFlowOutputs has no test coverage
## Gap

`InsufficientFlowOutputs` (declared at `src/error/ErrFlow.sol:31`) is reverted at `src/concrete/Flow.sol:207` when the deployer reports a flow output count below `flowMinOutputs` (which equals `MIN_FLOW_SENTINELS = 3` for `Flow.sol`, see `src/concrete/Flow.sol:134` and `src/interface/deprecated/v3/IFlowV3.sol:13`). No test asserts this revert path. Grep for `InsufficientFlowOutputs` in `test/` returns zero matches.

## Proposed test

Add to `test/src/concrete/Flow.construction.t.sol`:

```solidity
import {InsufficientFlowOutputs} from "src/error/ErrFlow.sol";

/// Reverts with the `InsufficientFlowOutputs` selector when the deployer
/// reports a flow output count below `MIN_FLOW_SENTINELS` (= 3) via the IO
/// byte string.
function testFlowConstructionInsufficientFlowOutputs(
    address expression,
    bytes memory bytecode,
    uint256[] memory constants
) external {
    // IO layout: byte(0) = inputs, byte(1) = outputs. Set outputs=2 (< 3) to trip the check.
    expressionDeployerDeployExpression2MockCall(expression, bytes(hex"00020001"));

    EvaluableConfigV3[] memory flowConfig = new EvaluableConfigV3[](1);
    flowConfig[0] = EvaluableConfigV3(DEPLOYER, bytecode, constants);

    vm.expectRevert(abi.encodeWithSelector(InsufficientFlowOutputs.selector));
    I_CLONE_FACTORY.clone(deployFlowImplementation(), abi.encode(flowConfig));
}
```

### PR DESCRIPTION
**Pull request by thedavidmeister**
## Summary

Pins the `flowInit` guard that rejects deployer-returned `flowOutputs < MIN_FLOW_SENTINELS` (= 3). Two tests:

- **Negative** (fuzzed): any `flowOutputs` in `[0, MIN_FLOW_SENTINELS)` reverts with `InsufficientFlowOutputs`.
- **Boundary**: `flowOutputs == MIN_FLOW_SENTINELS` is accepted (lowest valid value).

**Mutation verified**: inverting the check (`<` → `>=`) makes both tests fail; reverting passes.

Closes #311 #321.

## Test plan

- [x] `testFlowConstructionRevertsOnInsufficientFlowOutputs` — 100 fuzz runs
- [x] `testFlowConstructionAcceptsFlowOutputsAtMin` — 2048 fuzz runs
- [x] mutation: invert the check in `Flow.sol:206` → both tests fail

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **Tests**
  * Added validation tests to verify flow initialization rejects insufficient output parameters
  * Added boundary condition tests confirming minimum acceptable output values are properly handled

### CODE REVIEW

File: test/src/concrete/Flow.construction.t.sol [hunk 1]
```diff
@@ -16,6 +17,45 @@ contract FlowConstructionTest is FlowTest {
...
+
+        address impl = deployFlowImplementation();
+        vm.expectRevert(InsufficientFlowOutputs.selector);
+        I_CLONE_FACTORY.clone(impl, abi.encode(flowConfig));
+    }
+
+    /// `flowOutputs == MIN_FLOW_SENTINELS` is the lowest accepted value.
+    /// Pinning this boundary against regression complements the negative
+    /// test above.
+    function testFlowConstructionAcceptsFlowOutputsAtMin(
```
[the remark this answers is not in the document]
Comment (thedavidmeister) [reply test/src/concrete/Flow.construction.t.sol:45]: Rejecting. The pattern you reference doesn't exist:

- Revert test pins `forge-config: default.fuzz.runs = 100` — an override **down** from the project default to keep the slow `vm.assume` sweep cheap.
- Accept test has no override and inherits `foundry.toml`'s `fuzz.runs = 2048` already.

Adding `default.fuzz.runs = 2048` to the accept test is a no-op duplicate of the project default, not "matching the pattern". The boundary test is already running 2048 iterations.
